Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -4638,6 +4641,24 @@ RefPtr<DocumentLoader> FrameLoader::loaderForWebsitePolicies(CanIncludeCurrentDo
     return loader;
 }
 
+void FrameLoader::updateNavigationAPIEntries()
+{
+    if (!m_frame->document() || !m_frame->document()->settings().navigationAPIEnabled())
+        return;
+    if (auto domWindow = m_frame->document()->domWindow()) {
+        if (RefPtr page = m_frame->page()) {
+            if (RefPtr currentItem = page->backForward().currentItem()) {
+                Vector<Ref<HistoryItem>> historyItems;
+                for (int index = 0; index < (int)page->backForward().count(); index++) {
+                    if (auto item = page->backForward().itemAtIndex(index - page->backForward().backCount()))
+                        historyItems.append(*item);
+                }
+                domWindow->navigation().initializeEntries(*currentItem, historyItems);
```

Comment:
please add a `protectedNavigation()` getter to LocalDOMWindow which returns a `Ref<Navigation>` and use this here. We shouldn't be calling a member function on an object we don't hold a strong ref to.

Reply:
Done.